v0/destinations/hs/transform.js

```diff
@@ -324,6 +334,21 @@ const processRouterDest = async inputs => {
   const errorRespList = [];
   // using the first destination config for transforming the batch
   const { destination } = inputs[0];
+  // reduce the no. of calls for properties endpoint
+  let propertyMap;
+  const traitsFound = inputs.some(input => {
+    if (
+      input.message.traits ||
+      input.message.context.traits ||
+      input.message.properties
+    ) {
+      return true;
+    }
+    return false;
```

Comment:
The logic above is not properly replicated here. Let's refactor that original logic into a method and use it in two places.

Reply:
The above original logic extracts the traits and here I think we just need to add the check for the presence of traits.

Reply:
I can think of two reasons here for the re-factoring:
- Extracting traits (in general) is not a costly operation.
- Having this logic in two places is a potential risk going forward.

Reply:
refactored the logic into a method and used it in two different places.